### PR TITLE
Improve/different removed dynamic values

### PR DESCRIPTION
**Pull request by NSElvis**
I thought the structure of the values would look different so it could be parsed by the nested attributes pod.

Look [this](https://github.com/hyperoslo/Form/blob/improve/different-removed-dynamic-values/Tests/Tests/FORMCollectionViewDataSourceTests.m#L360) and [this](https://github.com/hyperoslo/Form/blob/improve/different-removed-dynamic-values/Tests/Tests/FORMCollectionViewDataSourceTests.m#L393) test

Also I found that removing a section isn't updating the other sections properly, [here](https://github.com/hyperoslo/Form/blob/improve/different-removed-dynamic-values/Tests/Tests/FORMCollectionViewDataSourceTests.m#L423)

### CODE REVIEW

**Comment by zenangst** [comment]
@NSElvis  Your tests are failing :scream:

**Comment by NSElvis** [comment]
@zenangst I know, I just implemented the tests but didn't added the implementation for the second one.

Working on it

**Comment by zenangst** [comment]
:+1:

File: Source/FORMDataSource.m [hunk 1]
```diff
@@ -533,7 +532,11 @@ - (void)fieldCell:(UICollectionViewCell *)fieldCell updatedWithField:(FORMField
                 foundParsed = [section.sectionID hyp_parseRelationship];
                 foundParsed.attribute = nil;
                 foundParsed.index = @(self.formsManager.removedValues.count);
-                [self.formsManager.removedValues setValue:[removedKeys copy] forKey:[foundParsed key]];
+
+                for (NSString *removedKey in removedKeys) {
+                    [self.formsManager.removedValues setValue:self.values[removedKey] forKey:removedKey];
+                    [self.formsManager.values removeObjectForKey:removedKey];
+                }
```
Comment (zenangst) [inline Source/FORMDataSource.m]: This looks a lot cleaner, god job!

**Pull request merged**